
/EkfCalNode:
    ros__parameters:
        debug_log_level: 2
        data_logging_on: true
        body_data_rate: 100.0
        sim_params:
            seed: 0.0
            use_seed: true
            max_time: 10.0
            no_errors: true
            number_of_runs: 10
            run_number: 0
            truth_type: "cyclic"
            pos_frequency: [0.3, 0.5, 0.7]
            ang_frequency: [0.2, 0.3, 0.4]
            pos_offset: [0.0, 0.0, 0.0]
            ang_offset: [0.0, 0.0, 0.0]
            ang_amplitude: 0.1
            pos_amplitude: 1.0

        filter_params:
            process_noise: [
                0.0, 0.0, 0.0, # Position
                0.0, 0.0, 0.0, # Velocity
                1.0e-2, 1.0e-2, 1.0e-2, # Acceleration
                0.0, 0.0, 0.0, # Angular Position
                1.0e-2, 1.0e-2, 1.0e-2, # Angular Velocity
                0.0, 0.0, 0.0, # Angular Acceleration
            ]

        imu_list:
            - "vn300"
            - "vn100"

        camera_list:
            # - "ace"

        tracker_list:
            - "orb"

        imu:
            vn300:
                use_for_prediction: false
                is_extrinsic: false
                is_intrinsic: false
                rate: 400.0
                topic: "/vn300/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-6
                ang_stability: 1.0e-6
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [0.0, 0.0, 0.0]
                    ang_error: [0.0, 0.0, 0.0]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
            vn100:
                use_for_prediction: false
                is_extrinsic: true
                is_intrinsic: false
                rate: 100.0
                topic: "/vn100/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-6
                ang_stability: 1.0e-6
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    ang_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]

        camera:
            ace:
                rate: 20.0
                topic:  "/my_camera/pylon_ros2_camera_node/image_raw"
                pos_c_in_b: [0.0, 0.0, 0.0]
                ang_c_to_b: [0.5, -0.5, 0.5, -0.5]
                variance: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
                tracker: "orb"
                sim_params:
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    ang_error: [1.0e-1, 1.0e-1, 1.0e-1]

        tracker:
            orb:
                feature_detector: 2
                descriptor_extractor: 1
                descriptor_matcher: 0
                detector_threshold: 10.0
                pixel_error: 1.0
                min_feature_distance: 1.0
                sim_params:
                    feature_count: 100
                    room_size: 10.0
